### ISSUE TITLE
sort-keys on ObjectPattern

### ISSUE THREAD

**Issue by benderTheCrime** [eslint/eslint#7776]
**What version are you using?**
3.12.1

**What did you do?**
Used the "sort-keys" rule with `{ "props": true }`

**What happened?**
Object patterns (destructures) were not sorted accordingly (asc)

**What did you expect to happen?**
Object patterns (destructures) should have been sorted accordingly (asc)

This is a request for additional functionality. It seems like it should be possible using the "sort-keys" rule to sort destructures. 

Yes, this may cause defaulted assignments to previous values in the destructure to be mis-re-ordered:
```JavaScript
const { bar, foo = bar } = baz;
```
I think that perhaps there should be a handler for this situation instead of disallowing sorting of ObjectPatterns.

**Comment by benderTheCrime** [comment]
It looks like this was addressed here: https://github.com/eslint/eslint/issues/6896 after a `TypeError` was thrown on sort-keys in a destructuring assignment and the fix was to ignore ObjectPatterns:
https://github.com/eslint/eslint/commit/4eecf6e11b6b5b04ac31be68a1474cf7371b57cb

I'm not convinced that this was the correct fix.

**Comment by benderTheCrime** [comment]
I've created a version of this rule that has supports destructuring and ignores keys assigned to previous keys here: https://github.com/benderTheCrime/eslint-plugin-benderthecrime/blob/master/lib/rules/sort-keys.js

**Comment by not-an-aardvark** [comment]
Thanks for the proposal. Sorry, it looks like this issue slipped past us.

I think it would be reasonable to have an option to sort `ObjectPattern` keys as well. To maintain backwards compatibility, it would have to be off by default.

I've added a :+1: to this issue. To be accepted, someone on the team will need to champion it, and it will need two more :+1:s from team members.

**Comment by not-an-aardvark** [comment]
Thanks for your interest in improving ESLint. Unfortunately, it looks like this issue didn't get consensus from the team, so I'm closing it. We define [consensus](http://eslint.org/docs/maintainer-guide/issues.html#consensus) as having three :+1:s from team members, as well as a team member willing to champion the proposal. This is a high bar by design -- we can't realistically accept and maintain every feature request in the long term, so we only accept feature requests which are useful enough that there is consensus among the team that they're worth adding.

Since ESLint is pluggable and can load custom rules at runtime, the lack of consensus among the ESLint team doesn't need to be a blocker for you using this in your project, if you'd find it useful. It just means that you would need to [implement the rule](http://eslint.org/docs/developer-guide/working-with-rules) yourself, rather than using a bundled rule that is packaged with ESLint.
